src/main/java/org/opensearch/knn/index/query/model/AlgoQueryParameters.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.index.query.model;
+
+/**
+ * Functional Interface for algorithm specific parameters.
+ * These parameters are passed to the algorithm implementation while processing a query
+ */
+public interface AlgoQueryParameters {}
```

Comment:
Should this be an interface or abstract class? I see this as an abstract class rather than interface.

Reply:
Abstract classes open up possibility of abusing it for code reuse. Inheritance should only be used to enforce behavior, Composition should be used for code reuse. So default is interface. Abstract classes have limited use cases and they should be used sparingly.

Functional interfaces like these are generally used to have bounds on the type of objects passed and that helps not have type `Object` in the function signature which can be easily abused.